### ISSUE TITLE
Using generate_cstr(true) breaks output format

### ISSUE THREAD

**Issue by gwenn** [rust-lang/rust-bindgen#3047]
To reproduce,
- checkout https://github.com/rusqlite/rusqlite
- in `libsqlite3-sys` folder, execute `upgrade.sh` script

Without the following patch, generated bindings are correctly formatted.
But not with:
```diff
diff --git a/libsqlite3-sys/build.rs b/libsqlite3-sys/build.rs
index f96d656..53778b7 100644
--- a/libsqlite3-sys/build.rs
+++ b/libsqlite3-sys/build.rs
@@ -533,6 +533,7 @@ mod bindings {
         let mut bindings = bindgen::builder()
             .default_macro_constant_type(bindgen::MacroTypeVariation::Signed)
             .disable_nested_struct_naming()
+           .generate_cstr(true)
             .trust_clang_mangling(false)
             .header(header.clone())
             .parse_callbacks(Box::new(SqliteTypeChooser));
```

**Comment by nyurik** [comment]
This is due to rustfmt not understanding the `c"..."` syntax -- which is only available with the 2021+ edition. The issue is that `bindgen` executes `rustfmt` **without** the `--edition` param, so `rustfmt` defaults to 2015. I think we can just add the edition parameter, setting it to 2021 if we generate the C-literals

**Comment by nyurik** [comment]
A workaround for this issue:  create a `rustfmt.toml` in the root of your project, and add `edition = "2021"` to it

**Comment by pvdrz** [comment]
> A workaround for this issue:  create a `rustfmt.toml` in the root of your project, and add `edition = "2021"` to it

indeed, if you use a feature only available on edition 2021, you have to use a toolchain with that edition enabled

**Comment by nyurik** [comment]
I think its a workaround because a user would expect this to work automagically - i.e. bindgen output should be proper in every environment - thus we really need to pass `--edition` param to rustfmt when needed. I may try to hack a patch for that

**Comment by pvdrz** [comment]
> I think its a workaround because a user would expect this to work automagically - i.e. bindgen output should be proper in every environment

In that case maybe we should unconditionally use 2018 as the default edition instead.

> thus we really need to pass `--edition` param to rustfmt when needed.

Yes but you can also pass the edition in your `rustfmt.toml`. I think this is basically the same issue as with the msrv. You have to set your edition everywhere (`Cargo.toml`, `rustfmt.toml` and `bindgen`) otherwise you'll have unexpected outputs.

**Comment by nyurik** [comment]
Is there ever a case when bindgen 2018 output can be formatted in some broken way by the 2021 edition formatter?  If not, we might as well pass the `--edition 2021` to the rustfmt if the MSRV is for the version that supports it.   The rest of the user code can still be using 2018 or anything else based on Cargo.toml - bindgen doesn't care about that

**Comment by nyurik** [comment]
https://github.com/rust-lang/rust-bindgen/pull/3063 fixes this issue
